agent/consul/state/catalog.go

```diff
@@ -736,34 +737,31 @@ func (s *Store) EnsureService(idx uint64, node string, svc *structs.NodeService)
 	return nil
 }
 
+var errCASCompareFailed = errors.New("compare-and-set: comparison failed")
+
 // ensureServiceCASTxn updates a service only if the existing index matches the given index.
 // Returns a bool indicating if a write happened and any error.
```

Comment:
```suggestion
// Returns an error if the write didn't happen and nil if write was successful.
```

Reply:
Aha, I forgot to fix the comment. Thank you for the edit! sounds good to me